File: monitor/internal/git.go
package internal

import (
	"fmt"
	"os/exec"
)

func RunGit(message string) error {
	err := runGitCommand([]string{"add", "public/data/"})
	if err != nil {
		return err
	}

	err = runGitCommand([]string{"commit", fmt.Sprintf("-m \"%s\"", message)})
	if err != nil {
		return err
	}

	err = runGitCommand([]string{"push", "origin"})
	if err != nil {
		return err
	}

	return nil
}

func runGitCommand(args []string) error {
	cmd := exec.Command("git", args...)

	err := cmd.Run()
	if err != nil {
		return err
	}
	return nil
}

File: monitor/src/git.ts
import { exec } from "child_process"

export default async function runGit(message: string): Promise<void> {
	await runGitCommand("add", "public/data/")

	await runGitCommand("commit", `-m '${message}'`)

	await runGitCommand("push", "origin")
}

function runGitCommand(...args: string[]): Promise<void> {
	return new Promise<void>((resolve, reject) => {
		exec(`git ${args.join(" ")}`, (err, stdout, stderr) => {
			if (err) {
				reject(err)
			}
			if (stderr) {
				reject(stderr)
			}

			resolve()
		})
	})
}
